app/src/main/java/org/greenstand/android/TreeTracker/signup/SignupViewModel.kt

```diff
@@ -59,17 +65,24 @@ class SignupViewModel(private val users: Users) : ViewModel() {
 
     fun updateEmail(email: String) {
         // TODO validate data and show errors if needed, after click
-        _state.value = _state.value?.copy(credential = Credential.Email().apply { text = email })
+        _state.value = _state.value?.copy(email = email)
+        _state.value = _state.value?.copy(isEmailValid = Validation.isEmailValid(email))
```

Comment:
For all the state updates, please update all params with a single copy method.

For example:
_state.value = _state.value?.copy(param1 = 1, param2 = 2, etc..)

Without that, you'll be updating the state once per each value.